test/sql/test_join/T/test_driving_table_selection

```diff
@@ -0,0 +1,10 @@
+-- name: test_driving_table_selection
+create table T (a int, b int, c int) properties("replication_num"="1");
+
+insert into T values(1,1,1),(2,2,2),(3,3,3);
+
+create table T0  properties("replication_num"="1")  as select * from T;
+create table T1  properties("replication_num"="1")  as select * from T;
+create table T2  properties("replication_num"="1")  as select * from T;
+
+explain select *  from T where T.a = (select a from T0) or T.b = (select b from T1) or T.c = (select c from T2);
```

Comment:
1. it's better to not `explain` this directly, you can refer `assert_explain_contains` in sql-tests
```
function: assert_explain_contains('SELECT COUNT(*) FROM partitions_multi_column_1 WHERE c1=1 AND TRUE', 'partitions=1/3')
```
2. it's better to add some specific data and check rewrite result fine.